hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeUpdateHandler.java

```diff
@@ -184,4 +196,42 @@ private Node.Builder updateNode(
         }
         return nodeBuilder;
     }
+
+    private void handleAccountIdOnlyUpdate(PreHandleContext context, Node existingNode) throws PreCheckException {
+        if (existingNode.hasAccountId()) {
+            // Allow signature from either admin key or existing account key
+            Key requiredKey =
+                    oneOf(existingNode.adminKey(), context.getKeyFromAccount(existingNode.accountIdOrThrow()));
+            context.requireKeyOrThrow(requiredKey, INVALID_SIGNATURE);
+        } else {
+            // No account ID exists, so only admin key signature is acceptable
+            context.requireKeyOrThrow(existingNode.adminKey(), INVALID_ADMIN_KEY);
+        }
+    }
+
+    private boolean onlyUpdatesAccountID(@NonNull final NodeUpdateTransactionBody op) {
+        return op.hasAccountId()
+                && !op.hasDescription()
+                && !op.hasAdminKey()
+                && op.gossipEndpoint().isEmpty()
+                && op.serviceEndpoint().isEmpty()
+                && !op.hasGossipCaCertificate()
+                && !op.hasGrpcCertificateHash()
+                && !op.hasDeclineReward()
+                && !op.hasGrpcProxyEndpoint();
+    }
+
+    /**
+     * Creates a threshold key with threshold 1 with the given keys.
+     * @param keysRequired keys required
```

Comment:
Nit:

```suggestion
     * @param acceptedKeys all keys that can individually authorize an operation
```

`keysRequired` makes it sound like all keys are required